### DISCUSSION TITLE
Extension http - POST requests with headers (Authorization)

### DISCUSSION THREAD

**Discussion by dkirilkin** [supabase/supabase#30353] [Questions]
### Questions:
1. How to use extension http to call POST requests with headers (jwt token in headers) right?
I think the problem is in the headers.
2. Could you get an example with POST requests with headers and Authorization?

First I test POST requests to another Supabase project as external system. But I get a 401 error (invalid API Key).
![Снимок экрана 2024-11-01 181141](https://github.com/user-attachments/assets/019a7efd-1329-4319-8926-519d3989bbab)

### What have I tried so far?

1. I called API in Postman. It works. I got 201 response.
```
curl -X POST 'https://hfjykeknmooymayncknw.supabase.co/rest/v1/api_incomings' \

-H "Content-Type: application/json" \

-H "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhmanlrZWtubW9veW1heW5ja253Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3MTIxNDkyMDUsImV4cCI6MjAyNzcyNTIwNX0.mbVHRg_0YAbvjcWL0WhKFWaeweKK6rTcPIIgdpvKgU8" \

-H "apikey: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhmanlrZWtubW9veW1heW5ja253Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3MTIxNDkyMDUsImV4cCI6MjAyNzcyNTIwNX0.mbVHRg_0YAbvjcWL0WhKFWaeweKK6rTcPIIgdpvKgU8" \

-H "Prefer: return=minimal" \

-d '{"name":"test6"}'
```

2. I executed SQL query in Supabase. I got 401 error (invalid API Key).
```
SELECT content

FROM http((

'POST',

'https://hfjykeknmooymayncknw.supabase.co/rest/v1/api_incomings',

ARRAY[

http_header('apikey', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhmanlrZWtubW9veW1heW5ja253Iiwicm9zZSI6ImFub24iLCJpYXQiOjE3MTIxNDkyMDUsImV4cCI6MjAyNzcyNTIwNX0.mbVHRg_0YAbvjcWL0WhKFWaeweKK6rTcPIIgdpvKgU8'),

http_header('Authorization', 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhmanlrZWtubW9veW1heW5ja253Iiwicm9zZSI6ImFub24iLCJpYXQiOjE3MTIxNDkyMDUsImV4cCI6MjAyNzcyNTIwNX0.mbVHRg_0YAbvjcWL0WhKFWaeweKK6rTcPIIgdpvKgU8')

],

'application/json', -- Указываем тип контента

jsonb_build_object('name', 'test8')::text -- Указываем тело запроса в формате JSON

) :: http_request);
```
**I think the problem is in**
```
ARRAY[

http_header('apikey', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVC......
```

4. Also I tried function http_post. 401 error.

5. I checked logs in Supabase. I can't find in it what problem was.

 [logs Supabase.json](https://github.com/user-attachments/files/17658343/logs.Supabase.json)

**Comment by tecspda** [comment]
I have same problem. This is right way (help for me): [function http_header(unknown, unknown) does not exist #29186](https://github.com/orgs/supabase/discussions/29186#discussion-7162544)

**Reply by GaryAustin1** [reply]
> I have same problem. This is right way (help for me): [function http_header(unknown, unknown) does not exist #29186](https://github.com/orgs/supabase/discussions/29186#discussion-7162544)

You probably need to set the search path to include extensions if you are calling this in a function.  Or use `extensions.http_header`.

**Comment by ghost** [comment]
same !!!!
